src/steps/resource-manager/policy/client.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Client,
+  iterateAllResources,
+} from '../../../azure/resource-manager/client';
+import { PolicyClient } from '@azure/arm-policy';
+import { PolicyAssignment } from '@azure/arm-policy/esm/models';
+
+export class AzurePolicyClient extends Client {
+  /**
+   * This operation retrieves the list of all policy assignments associated with the given subscription.
+   * The list includes all policy assignments associated with the subscription,
+   * including those that apply directly or from management groups that contain the given subscription,
+   * as well as any applied to objects contained within the subscription.
+   * @param callback A callback function to be called after retrieving a Policy Assignment
+   */
+  public async iteratePolicyAssignments(
+    callback: (s: PolicyAssignment) => void | Promise<void>,
+  ): Promise<void> {
+    const serviceClient = await this.getAuthenticatedServiceClient(
+      PolicyClient,
+    );
+    return iterateAllResources({
+      logger: this.logger,
+      serviceClient,
+      resourceEndpoint: {
+        list: async () => serviceClient.policyAssignments.list(),
+        listNext: async (nextPageLink) =>
+          serviceClient.policyAssignments.listNext(nextPageLink),
+      },
```

Comment:
If there are no arguments to the `serviceClient.policyAssignments.list()` function, you can simply pass as below:

```suggestion
      resourceEndpoint: serviceClient.policyAssignment,
```